### ISSUE TITLE
Unable to choose enabled project templates

### ISSUE THREAD

**Issue by GeekArthur** [eclipse/codewind#1351]
**Codewind version:** latest master
**OS:** MacOS

**Che version:** 7.3.1
**IDE extension version:** latest
**IDE version:** latest
**Kubernetes cluster:** OKD

**Description:**
I enable all project templates, but I can only choose OpenShift templates as the screenshot shown below. In my case, OpenShift templates are the initial templates that I choose for the first time project creation. Seems like the behavior is that we can only display the project templates that we initially choose when we start the workspace, all the project templates we enable afterwards can not display.
![Screen Shot 2019-12-03 at 3 04 39 PM](https://user-images.githubusercontent.com/15131537/70086645-c6c6db00-15e0-11ea-9f7c-f7d47368370c.png)

I shell into the theia container then run the following commands, result might prove the behavior.
```
bash-4.4$ ./cwctl --json --insecure templates repos list --conid local
[
        {
                "description": "Codewind project templates help you create containerized projects for various runtimes.",
                "url": "https://raw.githubusercontent.com/codewind-resources/codewind-templates/master/devfiles/index.json",
                "name": "Default templates",
                "id": "",
                "enabled": true,
                "protected": true,
                "projectStyles": [
                        "Codewind"
                ]
        },
        {
                "description": "The set of templates for new OpenShift projects in Codewind.",
                "url": "file:///codewind-workspace/.extensions/codewind-odo-extension/templates/index.json",
                "name": "OpenShift templates",
                "id": "",
                "enabled": true,
                "protected": true,
                "projectStyles": [
                        "OpenShift"
                ]
        },
        {
                "description": "Use Appsody in Codewind to develop applications with sharable technology stacks.",
                "url": "https://github.com/appsody/stacks/releases/latest/download/incubator-index.json",
                "name": "Appsody Stacks - incubator",
                "id": "incubator",
                "enabled": true,
                "protected": true,
                "projectStyles": [
                        "Appsody"
                ]
        }
]
bash-4.4$ curl -k https://localhost:9090/api/v1/templates
[
  {
    "label": "OpenShift Node.js",
    "description": "OpenShift Node.js sample application",
    "language": "nodejs",
    "url": "https://github.com/sclorg/nodejs-ex",
    "projectType": "odo",
    "projectStyle": "OpenShift",
    "source": "OpenShift templates"
  },
  {
    "label": "OpenShift Perl",
    "description": "OpenShift Perl sample application",
    "language": "perl",
    "url": "https://github.com/sclorg/dancer-ex",
    "projectType": "odo",
    "projectStyle": "OpenShift",
    "source": "OpenShift templates"
  },
  {
    "label": "OpenShift Python",
    "description": "OpenShift Python sample application",
    "language": "python",
    "url": "https://github.com/sclorg/django-ex",
    "projectType": "odo",
    "projectStyle": "OpenShift",
    "source": "OpenShift templates"
  }
]bash-4.4$ 
```

**Comment by GeekArthur** [comment]
Another folk @maysunfaisal can see this issue as well, add stopship label since user can’t choose templates that they enable with this issue.

/priority stopship

**Comment by malincoln** [comment]
@tobespc pls review and assign.  Thanks

**Comment by makandre** [comment]
This is possibly caused by https://github.com/eclipse/codewind/pull/1292

On quick glance I don't see `projectTemplatesNeedsRefresh` getting set to `true` when a repo is enabled.

**Comment by GeekArthur** [comment]
I think Andrew is right, we should set `projectTemplatesNeedsRefresh` to `true` when calling PATCH /api/v1/batch/templates/repositories since seems like PATCH api/v1/batch/templates/repositories is responsible for handling repo enable, below are logs when enable repo from IDE

```
Extension-Host:[INFO: 17:06:41.815 ManageTemplateReposCmd.js:96]: Enable/Disable repos: { repos:
   [ { repoID:
        'https://raw.githubusercontent.com/codewind-resources/codewind-templates/master/devfiles/index.json',
       enable: true } ] }
Extension-Host:[DBUG: 17:06:41.815 Requester.js:355]: Doing PATCH request to https://localhost:9090/api/v1/batch/templates/repositories
```

**Comment by malincoln** [comment]
@GeekArthur are you able to try the fix and confirm?  I see it was merged a few days ago.

**Comment by GeekArthur** [comment]
@malincoln Me and Andrew find the root cause, but we are not the owner of this PR https://github.com/eclipse/codewind/pull/1292 that potentially cause the problem and we are not familiar with the part of Portal's code. To avoid inject a simple fix and bring more issues up, @james-wallis can you take a look at this and fix this since you create the PR?

**Comment by james-wallis** [comment]
PR created for this issue https://github.com/eclipse/codewind/pull/1428

**Comment by james-wallis** [comment]
@GeekArthur would you be able to review the PR?

**Comment by malincoln** [comment]
@GeekArthur pls verify and close.  Thanks

**Comment by GeekArthur** [comment]
@james-wallis Verified, the issue has been resolved, thanks. We should backport the fix to 0.7.0 as well, once it's done we can close this issue.

**Comment by james-wallis** [comment]
PR Opened for 0.7.0 branch https://github.com/eclipse/codewind/pull/1474

**Comment by james-wallis** [comment]
@GeekArthur PR has been merged into 0.7.0 branch, can you verify and close please.

**Comment by GeekArthur** [comment]
Verified 0.7.0, it works, thanks @james-wallis, close the issue.
